src/core/Lockfile_target.ml

```diff
@@ -1,25 +1,76 @@
-(* A lockfile to be scanned by a supply chain rule.
-   Should contain a list of all dependencies used by
-   a project, at exact versions. Can be attached to a code target,
-   or a target by itself.
-*)
 type t = {
-  lockfile : Fpath.t;
-  lockfile_kind : Lockfile_kind.t;
-  lazy_lockfile_content : string lazy_t;
-  lazy_lockfile_ast_and_errors : Dependency.t list lazy_t;
-  manifest_target : manifest_target option;
+  source : Source.t;
+      (** The source of the data as is relevant to the user. This could be, e.g., a
+      relative (from the project root) path to a file, a git object and
+      associated information, or anything else a Source.t can designate.
+
+      This should be used when reporting a location to the user. *)
+  internal_path_to_content : Fpath.t;
```

Comment:
I still see those source: xxx; internal_path_to_content: xxx everywhere, which is bad smell.

Reply:
Addressed in fc35371b4